src/dash-table/dash/DataTable.js

```diff
@@ -454,34 +442,62 @@ export const propTypes = {
     merge_duplicate_headers: PropTypes.bool,
 
     /**
-     * `n_fixed_columns` will "fix" the set of columns so that
+     * `fixed_columns` will "fix" the set of columns so that
      * they remain visible when scrolling horizontally across
-     * the unfixed columns. `n_fixed_columns` fixes columns
-     * from left-to-right, so `n_fixed_columns=3` will fix
-     * the first 3 columns.
+     * the unfixed columns. `fixed_columns` fixes columns
+     * from left-to-right.
+     *
+     * If `headers` is False, no columns are fixed.
+     * If `headers` is True, all operation columns (see `row_deletable` and `row_selectable`)
+     * are fixed. Additional data columns can be fixed by
+     * assigning a number to `data`.
+     *
+     * Defaults to `{ headers: False }`.
      *
      * Note that fixing columns introduces some changes to the
      * underlying markup of the table and may impact the
      * way that your columns are rendered or sized.
      * View the documentation examples to learn more.
      */
-    n_fixed_columns: PropTypes.number,
+    fixed_columns: PropTypes.oneOfType([
+        PropTypes.exact({
+            headers: PropTypes.oneOf([false]),
+            data: PropTypes.oneOf([0])
+        }),
+        PropTypes.exact({
+            headers: PropTypes.oneOf([true]),
+            data: PropTypes.number
+        })
```

Comment:
I guess as specified here (no `.isRequired` anywhere) someone could pass `{data: 2}` and it would be a valid value. And in fact that seems kind of nice from the user's side, *if* it effectively causes `headers: true` - does it?

Reply:
Alternatively I guess in the second form only:
`headers: PropTypes.oneOf([true]).isRequired`
would make `{data: 2}` invalid, and you need to explicitly turn on `headers` first if you want fixed data rows.

Reply:
Nope, doesn't do anything thing like that, updating for headers to be required.

Reply:
Sure. If we decide later to allow `{data: 2}` with implied `headers: true` we can always add that in.